recipes/util-is/src/workflow.ts

```diff
@@ -0,0 +1,218 @@
+            }
+        }
+
+        // require namespace: const util = require('node:util')
+    const reqNs = getRequireNamespaceIdentifier(node);
+        if (reqNs) namespaceBindings.add(reqNs.text());
+    }
+
+    // Mark non-is util usages for any namespace binding discovered
+    for (const ns of namespaceBindings) {
+        const usages = rootNode.findAll({ rule: { pattern: `${ns}.$METHOD($$$)` } });
+        for (const usage of usages) {
+            const methodMatch = usage.getMatch('METHOD');
+            if (methodMatch) {
+                const methodName = methodMatch.text();
+                if (!(methodName in replacements)) nonIsMethodsUsed.add(methodName);
+            }
+        }
+    }
+
+    // Resolve local bindings for each util.is* and replace invocations
+    const localRefsByMethod = new Map<string, Set<string>>();
+    for (const [method] of Object.entries(replacements)) {
+        localRefsByMethod.set(method, new Set());
+        for (const node of importOrRequireNodes) {
+            const resolved = resolveBindingPath(node, `$.${method}`);
+            if (resolved) localRefsByMethod.get(method)!.add(resolved);
+        }
+    }
+
+    for (const [method, replacement] of Object.entries(replacements)) {
+        const refs = localRefsByMethod.get(method)!;
+        for (const ref of refs) {
+            const calls = rootNode.findAll({ rule: { pattern: `${ref}($ARG)` } });
+            for (const call of calls) {
+                const arg = call.getMatch('ARG');
+                if (!arg) continue;
+                const newCallText = replacement(arg.text());
+                edits.push(call.replace(newCallText));
+                usedMethods.add(method);
+            }
+        }
+    }
+
+    if (!edits.length) return null;
+
+    // Clean up unused imports using removeBinding
+    const allIsMethods = [
+        'isArray', 'isBoolean', 'isBuffer', 'isDate', 'isError', 'isFunction',
+        'isNull', 'isNullOrUndefined', 'isNumber', 'isObject', 'isPrimitive',
+        'isRegExp', 'isString', 'isSymbol', 'isUndefined'
+    ];
+
+    const importStatements = getNodeImportStatements(root, 'util');
+    for (const importNode of importStatements) {
+    const hasNamespace = Boolean(importNode.find({ rule: { kind: 'namespace_import' } }));
+    const namedImportSpecifiers = importNode.findAll({ rule: { kind: 'import_specifier' } });
+    const hasNamed = namedImportSpecifiers.length > 0;
+    const defaultIdentifier = getDefaultImportIdentifier(importNode);
+
+        // If all named specifiers are util.is* and there is no default or namespace, drop whole line
+        if (
+            hasNamed && !defaultIdentifier && !hasNamespace &&
+            namedImportSpecifiers.every((spec) => {
+                const firstIdent = spec.find({ rule: { kind: 'identifier' } });
+                const name = firstIdent?.text();
+                return name ? allIsMethods.includes(name) : false;
+            })
```

Comment:
nit

```suggestion
            hasNamed
            && !defaultIdentifier
            && !hasNamespace &&
            namedImportSpecifiers.every(hasAnyOtherNamedImports)
```

extracting `hasAnyOtherNamedImports` into a util declared in the module's root (it doesn't need to get re-constructed on every run).

the return of `hasAnyOtherNamedImports` can be simplified from a ternary to a binary:

```
return match && allIsMethods.includes(name);
```